.github/workflows/nightly.yml

```
@@ -0,0 +1,117 @@
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: nightly
+
+on:
+  schedule:
+    - cron: "0 4 * * *"
+
+jobs:
+  license:
+    name: License Scanner
+    uses: ./.github/workflows/license.yml@develop
+
+  tests:
+    name: Unit and Functional Tests
+    uses: ./.github/workflows/tests.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  benchmarks:
+    name: Benchmarks
+    uses: ./.github/workflows/benchmark.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  changelog:
+    name: Assign Version
+    needs:
+      - license
+      - tests
+    uses: ./.github/workflows/changelog.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  push_to_master:
+    name: Push to Master
+    runs-on: ubuntu-latest
+    needs:
+      - changelog
+    steps:
+      - name: Get latest release
+        id: query-tags
+        uses: fjogeleit/http-request-action@v1
+        with:
+          url: "https://api.github.com/repos/AgnostiqHQ/covalent/tags"
+          method: "GET"
+
+      - name: Parse latest tag
+        id: get-latest-tag
+        run: |
+          # This assumes that the response from the API is sorted in decreasing order (thus the first element is the latest tag)
+          latest_tag=${{ fromJSON(steps.query-tags.outputs.response)[0].name }}
+          echo "::set-output name=tag::${latest_tag}"
+
+      - name: Push to master
+        id: push
+        run: |
+          MASTER_VERSION="$(echo ${{ steps.get-latest-tag.outputs.tag }} | cut -c2- )"
+          VERSION="$(cat ./VERSION)"
+          release=false
+          if [ "$MASTER_VERSION" = "$VERSION" ] ; then
+            echo "$VERSION has been previously released."
+          elif dpkg --compare-versions $VERSION 'gt' '0.177.0' ; then
+            git config user.name "CovalentOpsBot"
+            git config user.email "covalentopsbot@users.noreply.github.com"
+            git remote set-url origin https://${{ secrets.COVALENT_OPS_BOT_TOKEN }}@github.com/AgnostiqHQ/covalent.git
+            git push origin HEAD:master
+            release=true
+          else
+            echo "We cannot release versions less than 0.177.0."
+          fi
+          echo "RELEASE=$release" >> $GITHUB_ENV
+          echo "::set-output name=release::$release"
+
+  release:
+    name: Create Release
+    needs:
+      - push_to_master
+    uses: ./.github/workflows/release.yml@develop
+    with:
+      prerelease: true
+    secrets: inherit # pragma: allowlist secret
+
+  notify:
+    name: Notify Slack
+    needs:
+      - release
```

```suggestion
    needs: release
```
this isn't necessary, I just thought it looked a little cleaner